frontend/www/js_templates/phonebook.js

```diff
@@ -165,9 +188,102 @@ async function loadEntityList(parentEntity=null) {
+          edit_entity_btn.onclick = function(){
+            window.location.href = `[% path %]new/index.cgi?action=edit_entity&entity_id=${entityID}`;
+          };
+
+          add_entity_btn.style.display = 'block'; 
+          add_entity_btn.onclick = function(){
+            window.location.href = `[% path %]new/index.cgi?action=add_entity&entity_id=${entityID}`;
+          };
+
+          let user_list = document.querySelector('#user-list');
+          user_list.innerHTML ="";
+          var search = document.createElement("INPUT");
+          search.setAttribute('type','text');
+          search.onkeyup = function () { filterFunction()};
+          search.id = "user-search";
+          search.placeholder = "Search...";
+          user_list.appendChild(search);
+
+          add_user_btn.style.display = 'block';
+          let users_url = `[% path %]services/data.cgi?action=get_users`;
+          const users_resp = await fetch(users_url, {method:'get', credentials:'include'});
+          var users = await users_resp.json();
+          users = users['results'];
+          let i =0;
+          for (i =0 ; i < users.length; i++){
+            var ele = document.createElement("A");
+            let user_id = users[i]['user_id'];
+            if (!contact_ids.includes(user_id)){
+              ele.onclick = function(){ add_user(user_id, entityID)};
+              var t = document.createTextNode(users[i]['first_name'] + " "  +users[i]['family_name']);
+              ele.appendChild(t);
+              user_list.appendChild(ele);
+            }
+          }
+    }
+    else{
+      add_entity_btn.style.display = 'none'; 
+      edit_entity_btn.style.display = 'none'; 
+      add_user_btn.style.display = 'none';
+    } 
+}
+
+function showDropdown()   {
+  document.getElementById("user-list").classList.toggle("show");
+}
+
+function filterFunction() {
+  var input, filter, ul, li, a, i;
+  input = document.getElementById("user-search");
+  filter = input.value.toUpperCase();
+  div = document.getElementById("user-list");
+  a = div.getElementsByTagName("a");
+  for (i = 0; i < a.length; i++) {
+    txtValue = a[i].textContent || a[i].innerText;
+    if (txtValue.toUpperCase().indexOf(filter) > -1) {
+      a[i].style.display = "";
+    } else {
+      a[i].style.display = "none";
+    }
+  }
 }
+
+// Add user to entity
+async function add_user(user_id, entityID){
+  const url = `[% path %]services/entity.cgi?action=add_user&entity_id=${entityID}&user_id=${user_id}`;
+  const response = await fetch(url, {method:'get', credentials:'include'});
+  const data = await response.json();
+  await loadEntityList(entityID);
+ }
+
+async function remove_user(user_id, entityID){
+  const url = `[% path %]services/entity.cgi?action=remove_user&entity_id=${entityID}&user_id=${user_id}`;
+  const response = await fetch(url, {method:'get', credentials:'include'});
+  const data = await response.json;
+
+  await loadEntityList(entityID);
+}
+
```

Comment:
It would be preferable to have both fetch's in L275-L288 to be abstracted out into their own funcs in `js_templates/api/entity.js`.